lib/VolSurface/Utils.pm

```diff
@@ -89,16 +89,20 @@ sub get_delta_for_strike {
         croak "Arg $_ is undef at get_delta_for_strike" unless defined $args->{$_};
     }
 
+    $new_args{forward} = $new_args{t} >= 1 ? 1 : 0;
+
     my ($K, $sigma, $t, $S, $r, $q, $premium_adjusted) =
         ($new_args{strike}, $new_args{atm_vol}, $new_args{t}, $new_args{spot}, $new_args{r_rate}, $new_args{q_rate}, $new_args{premium_adjusted});
 
     my $delta;
     if ($premium_adjusted) {
+        my $forward_adj = $new_args{forward} ? 1 : exp(-1 * $new_args{r_rate} * $new_args{t});
```

Comment:
Turns out this is actually supposed to be `$new_args{forward} ? exp($new_args{q_rate} * $new_args{t}) : 1`. Note the lack of negative next to `q_rate`